src/query-builder/SelectQueryBuilder.ts

```diff
@@ -1819,10 +1819,11 @@ export class SelectQueryBuilder<Entity> extends QueryBuilder<Entity> implements
 
         if (rawResults.length > 0) {
 
+            const queryRunner = this.obtainQueryRunner();
             // transform raw results into entities
             const rawRelationIdResults = await relationIdLoader.load(rawResults);
             const rawRelationCountResults = await relationCountLoader.load(rawResults);
-            const transformer = new RawSqlResultsToEntityTransformer(this.expressionMap, this.connection.driver, rawRelationIdResults, rawRelationCountResults);
+            const transformer = new RawSqlResultsToEntityTransformer(this.expressionMap, this.connection.driver, rawRelationIdResults, rawRelationCountResults, queryRunner);
```

Comment:
take a look how in all other places where `this.obtainQueryRunner()` is called resulted query runner is released if it was created by query runner. And it should be done only once because once query runner is released you can't use it anymore.

I think proper solution here is to make queryRunner inside `RawSqlResultsToEntityTransformer` constructor optional and pass `this. queryRunner` here.

Sorry for being annoying about query runners here, but they are very dangerous if are not used properly and can lead to leaks and hard-to-debug bugs.

Reply:
yeah no worries. still getting the hang of the setup.

Reply:
You still have an issue here (use this.queryRunner and remove `const queryRunner = this.obtainQueryRunner();`). Maybe you'll success with tests if you fix this one.

Reply:
That was it. Thanks!